### DISCUSSION TITLE
Running Release build error

### DISCUSSION THREAD

**Discussion by borioan** [Macad3D/Macad3D#5] [General]
Hi!
I encounter a strange issue when I want to launch Macad from VisualStudio - Release build. 
I indicate the steps that have been taken:
I'm using the latest version of Visual Studio - version 16.8.3, Windows 10 Pro - 20H2
Before starting the Macad source compilation process I have tested Macad installing the exe and play with the software
While the installed version still exists on the computer, I started building the app from source following the steps indicated on the main page. After finishing the build process using > build all, I opened Macad3D.sln using VisualStudio, and I see that it is generating an error: it can not load Macad.UserGuide project (incompatible) as you can see in the picture below. It is not clear to me what this project does, but it seems not to affect the whole build of the app.  Ignoring this error I built the entire solution again using this time VisualStudio menu-> Build-> Rebuild Solution (Release build). After the complete build process was successfully, I try to run the app from VisualStudio Debug menu->Start Without Debugging (Release build) but the execution end up with the error in the picture.
I can perfectly build the whole solution under Debug build and open it directly from VisualStudio (Debug/F5 or Start Without Debugging/Ctrl+F5) but for the Release build, the error in the picture below is generated.. I do not know how to proceed with Release build, I uninstall the Macad app installed before from the exe but still have the same problem. The picture is generate from a very fast print screen immediately after launch the app from VisualStudio Debug menu->Start Without Debugging. As you can see, the WelcomeDialog is displayed until the error occurs..

![Release_error](https://user-images.githubusercontent.com/31766527/103345547-5831a680-4a9a-11eb-91b7-8c848c54aefd.jpg)

Any idea is welcome!!

**Comment by dipts** [comment]
I'm sorry, but I cannot reproduce the crash. Can you look at the call stack and see where what goes wrong?

**Reply by borioan** [reply]
Sure, but allow me some time to understand how to proceed, I have to learn! Also I will test the problem on other computer to see if it is repeating. 
Keep in touch!

**Comment by dipts** [comment]
I found out. The binary download of OCCT was compiled with additional dependencies that are not copied by the build system. A simple copy of the DLLs from FreeImage, OpenVR and FFMPEG into the Bin\Release directory is sufficient.

I will fix that.

**Reply by borioan** [reply]
Excellent!!! Meanwhile I have discovered some bugs, but I must understand how to use git/github before making a pull request to the current repository!

**Comment by dipts** [comment]
Should be fixed with https://github.com/Macad3D/Macad3D/commit/8129f4e428a9ae3f9064875e4e5c16dc01d1ee93. 

Please note that OCCT 7.5.0 is now used. The good news is that the distribution no longer needs to be imported. See updated [readme](https://github.com/Macad3D/Macad3D/blob/main/Readme.md).

I hope the release build works for you now.

**Comment by borioan** [comment]
Release version works perfectly on old version 7.4.0 - from VisualStudio Debug menu->Start Without Debugging after replacing the dlls !!
Excellent news - about using the last version of OCCT - 7.5.0!!!! 
I will test it!
May I ask you for a private e-mail? 
Thank you!!

**Reply by dipts** [reply]
> May I ask you for a private e-mail?

Sure, write to info@macad3d.net.

**Comment by borioan** [comment]
Perfect!!
Thank you for your very fast responses!

**Comment by borioan** [comment]
Everything works just perfect in both situations - Debug and Build - with the new version of OCCT - 7.5.0!!
You did a great job!
E-mail sent! :-)
